src/device/device_io_hid.cpp

```diff
@@ -44,8 +44,15 @@ namespace hw {
     
     static std::string safe_hid_error(hid_device *hwdev) {
       if (hwdev) {
-        const char* error_str = (const char*)hid_error(hwdev);
-        return  std::string(error_str == nullptr ? "Unknown error" : error_str);
+        const wchar_t* error_wstr = hid_error(hwdev);
+        if (error_wstr == nullptr)
+        {
+          return "Unknown error";
+        }
+        std::mbstate_t state{};
+        std::string error_str(std::wcsrtombs(nullptr, &error_wstr, 0, &state) + 1, 0);
+        std::wcsrtombs(&error_str[0], &error_wstr, error_str.size(), &state);
```

Comment:
If the call fails, this calls &error_str[0] on a string of size 0 (return -1, adds 1). I'm not sure this is valid.

Reply:
Added explicit error check